cmd/cmd_renew.go

```diff
@@ -119,7 +129,16 @@ func renewForDomains(ctx *cli.Context, client *lego.Client, certsStorage *Certif
 
 	cert := certificates[0]
 
-	if !needRenewal(cert, domain, ctx.Int("days")) {
+	var ariSaysRenew bool
+	if ctx.Bool("ari-enable") {
+		if len(certificates) < 2 {
+			log.Warnf("[%s] Certificate bundle does not contain issuer, cannot use the renewalInfo endpoint", domain)
+		} else {
+			ariSaysRenew = needRenewalARI(ctx, certificates[0], certificates[1], domain, client)
```

Comment:
It feels weird for me for the sleep to happen inside a function that returns a boolean. Feels too side-effect-y. Maybe this is a comment you want to save for the actual PR, but I'd personally prefer it if this helper returned the suggested timestamp, and then the check against ari-willing-to-sleep happened out here in the straight-line code.

Reply:
Fair. I was mostly looking to de-duplicate some code but I agree that it's sorta wonky.